rfd/0077-support-dump.md

```diff
@@ -0,0 +1,252 @@
+possible to grab those, any number of other kubernetes objects that are
+available should be retrieved. Secrets and other sensitive information should
+be redacted.
+
+Grabbing all possible resources in the relevant namespace is ideal because the
+exact methods used to deploy teleport in kubernetes vary. Some people use our
+helm charts, some people use their own method.
+
+This directory structure might look something like this:
+
+```
+└── kube
+    ├── kubectl_get_all.yaml
+    ├── kubectl_get_bindings.yaml
+    ├── kubectl_get_certificaterequests.cert-manager.io.yaml
+    ├── kubectl_get_certificates.cert-manager.io.yaml
+    ├── kubectl_get_challenges.acme.cert-manager.io.yaml
+    ├── kubectl_get_configmaps.yaml
+    ├── kubectl_get_controllerrevisions.apps.yaml
+    ├── kubectl_get_cronjobs.batch.yaml
+    ├── kubectl_get_crontabs.stable.example.com.yaml
+    ├── kubectl_get_daemonsets.apps.yaml
+    ├── kubectl_get_deployments.apps.yaml
+    ├── kubectl_get_endpoints.yaml
+    ├── kubectl_get_endpointslices.discovery.k8s.io.yaml
+    ├── kubectl_get_events.events.k8s.io.yaml
+    ├── kubectl_get_events.yaml
+    ├── kubectl_get_horizontalpodautoscalers.autoscaling.yaml
+    ├── kubectl_get_ingresses.extensions.yaml
+    ├── kubectl_get_ingresses.networking.k8s.io.yaml
+    ├── kubectl_get_issuers.cert-manager.io.yaml
+    ├── kubectl_get_jobs.batch.yaml
+    ├── kubectl_get_leases.coordination.k8s.io.yaml
+    ├── kubectl_get_limitranges.yaml
+    ├── kubectl_get_localsubjectaccessreviews.authorization.k8s.io.yaml
+    ├── kubectl_get_networkpolicies.networking.k8s.io.yaml
+    ├── kubectl_get_orders.acme.cert-manager.io.yaml
+    ├── kubectl_get_persistentvolumeclaims.yaml
+    ├── kubectl_get_persistentvolumes.yaml
+    ├── kubectl_get_poddisruptionbudgets.policy.yaml
+    ├── kubectl_get_pods.metrics.k8s.io.yaml
+    ├── kubectl_get_pods.yaml
+    ├── kubectl_get_podtemplates.yaml
+    ├── kubectl_get_replicasets.apps.yaml
+    ├── kubectl_get_replicationcontrollers.yaml
+    ├── kubectl_get_resourcequotas.yaml
+    ├── kubectl_get_rolebindings.rbac.authorization.k8s.io.yaml
+    ├── kubectl_get_roles.rbac.authorization.k8s.io.yaml
+    ├── kubectl_get_secrets.yaml
+    ├── kubectl_get_securitygrouppolicies.vpcresources.k8s.aws.yaml
+    ├── kubectl_get_serviceaccounts.yaml
+    ├── kubectl_get_services.yaml
+    ├── kubectl_get_statefulsets.apps.yaml
+    ├── kubectl_logs_teleport_ct-bb79b9bdb-xbsmq_alpine.log
+    ├── kubectl_logs_teleport_etcd-865c899bf6-xlzcl_etcd.log
+    └── kubectl_logs_teleport_teleport-8657cd5579-nv48j_teleport.log
+```
+
+#### Miscellaneous Considerations
+
+When a support dump is invoked, it should print minimal progress information.
+There should be no surprises such as timing out with a failure after five
+minutes. It should instead be obvious that the support dump is working on
+something and fail quickly if it is going to fail. Users can be told to opt in
```

Comment:
Similar to above, a lot of these goals (minimal progress, failing fast, not timing out) conflict with the goal of collecting "anything that can possibly be collected"

Reply:
yes indeed-- a tunable balance is the real goal. The support dump should have as much capability as is possible, but have the ability to fail fast and advise that the expected file size is large or the time to do the process will be too long. For really nasty cases, it's handy to be able to turn off the warnings and really get everything no matter how long it will take.